src/opendr/perception/facial_expression_recognition/landmark_based_facial_expression_recognition/algorithm/datasets/landmark_extractor.py

```diff
@@ -0,0 +1,98 @@
+"""
+Modified based on: Adrian Rosebrock, Facial landmarks with dlib, OpenCV, and Python, PyImageSearch,
+https://www.pyimagesearch.com/2017/04/03/facial-landmarks-dlib-opencv-python/, accessed on 27 August 2020
+"""
```

Comment:
This code is modified based on a code in a blog post. I followed the author's instruction to cite his code (referring to his block post and mentioning the day I accessed it, etc). It is not available on Github and I don't see any license for it. Do I need to ask for permission from the author?

Reply:
I am not sure how this goes, since this is a tutorial meant to demonstrate to other uses how to develop code. In this sense,  I think it is probably reasonable to assume that the author would probably expect that her/his code would be used for teaching other how to code - and as result - for some tasks some similarities might arise inevitable. However, I do not know if this is a valid argument (I can find arguments against this) and I do not know the laws/standard practices regarding this aspect. Therefore, in my view  the most safe and clean solution would be to ask if the author of the tutorial is ok with using the code that you wrote when you followed this tutorial in an open source project under the Apache 2 license.

Reply:
The code is implemented from scratch and only one function is modified based on this blog post. BTW, I asked the author and updated the license respectively. 
The branch is ready to be merged, but the Skeleton-based_human_action_recognition branch needs to be merged before this branch.